src/Sql/Sql/ServerTrustGroup/Services/AzureSqlServerTrustGroupAdapter.cs

```diff
@@ -0,0 +1,113 @@
+
+namespace Microsoft.Azure.Commands.Sql.ServerTrustGroup.Services
+{
+	/// <summary>
+	/// Adapter for Server Trust Group operations
+	/// </summary>
+	public class AzureSqlServerTrustGroupAdapter
+	{
+		private AzureSqlServerTrustGroupCommunicator Communicator { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Azure profile
+        /// </summary>
+        public IAzureContext Context { get; set; }
+
+        /// <summary>
+        /// Constructs a database adapter
+        /// </summary>
+        /// <param name="profile">The current azure profile</param>
+        /// <param name="subscription">The current azure subscription</param>
+        public AzureSqlServerTrustGroupAdapter(IAzureContext context)
+        {
+            Context = context;
+            Communicator = new AzureSqlServerTrustGroupCommunicator(Context);
+        }
+
+        public AzureSqlServerTrustGroupModel GetServerTrustGroup(string resourceGroupName, string locationName, string serverTrustGroupName)
+		{
+            var res = Communicator.Get(resourceGroupName, locationName, serverTrustGroupName);
+            return CreateServerTrustGroupModelFromResponse(res);
+		}
+
+        public List<AzureSqlServerTrustGroupModel> ListServerTrustGroupsByInstance(string resourceGroupName, string serverTrustGroupName)
+		{
+            IPage<Management.Sql.Models.ServerTrustGroup> entities = Communicator.ListGroupsByInstance(resourceGroupName, serverTrustGroupName);
+            List<AzureSqlServerTrustGroupModel> models = new List<AzureSqlServerTrustGroupModel>();
+            foreach(Management.Sql.Models.ServerTrustGroup entity in entities)
+			{
+                models.Add(CreateServerTrustGroupModelFromResponse(entity));
+			}
+
+            return models;
+		}
+
+        public List<AzureSqlServerTrustGroupModel> ListServerTrustGroupsByLocation(string resourceGroupName, string location)
+		{
+            IPage<Management.Sql.Models.ServerTrustGroup> entities = Communicator.ListGroupsByLocation(resourceGroupName, location);
+            List<AzureSqlServerTrustGroupModel> models = new List<AzureSqlServerTrustGroupModel>();
+            foreach (Management.Sql.Models.ServerTrustGroup entity in entities)
+            {
+                models.Add(CreateServerTrustGroupModelFromResponse(entity));
+            }
+
+            return models;
+        }
+
+        public AzureSqlServerTrustGroupModel CreateServerTrustGroup(AzureSqlServerTrustGroupModel model)
+		{
+            Management.Sql.Models.ServerTrustGroup parameters = new Management.Sql.Models.ServerTrustGroup();
+            parameters.GroupMembers = new List<ServerInfo>();
+            parameters.TrustScopes = model.TrustScope;
+            foreach(String member in model.GroupMembers)
+            {
+                parameters.GroupMembers.Add(new ServerInfo($"/subscriptions/{Context.Subscription.Id}/resourceGroups/{model.ResourceGroupName}/providers/Microsoft.Sql/managedInstances/{member}"));
```

Comment:
model.ResourceGroupName represents the resource group of Server trust group. The resource group of each group member can differ.

Can you extract the resource group of MI from model.GroupMembers, or any other way?

Reply:
Same goes for subscription, we want to enable cross-subscription scenarios which means that two MIs that belong to the same STG can be from different subscriptions.